### DISCUSSION TITLE
Sending Messages in CommunityToolkit

### DISCUSSION THREAD

**Discussion by Elencon** [CommunityToolkit/dotnet#344] [Q&A] [answered]
Hello 
Visual Studio 17.2.6 
 and "CommunityToolkit.Mvvm" Version="7.1.2" 
 1) I created WPF Application project
 2) Fixed compilation problems with Source Generators with workaround:
 https://github.com/dotnet/wpf/pull/6680#issuecomment-1183455941
 3)Here is MyViewModel code
 namespace CommunityToolkit;

public class TestMessage
{
}

public class MyViewModel : ObservableRecipient,
    IRecipient<TestMessage>
{
    private string _first_name;
    public string FirstName
    {
        get => _first_name;
        set => SetProperty(ref _first_name, value);
    }

    private string _last_name;
    public string LastName
    {
        get => _last_name; 
        set => SetProperty(ref _last_name, value);
    }

    public MyViewModel()
    {
        FirstName = "John";
        LastName = "Doe";
        SendMessage();
    }

    public void SendMessage()
    {
        WeakReferenceMessenger.Default.Send(new TestMessage());
    }
    public void Receive(TestMessage m)
    {
        int a = 0;
    }
}

4) I see that SendMessage is executed in debugger, but Receive function is not called
Here is my MainWindow.xaml

<Window x:Class="CommunityToolkit.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:local="clr-namespace:CommunityToolkit"
        mc:Ignorable="d"
        Title="MainWindow" Height="450" Width="800">
    <Grid>
        <Grid.ColumnDefinitions>
        <ColumnDefinition Width="Auto"/>
        <ColumnDefinition Width="*"/>
        </Grid.ColumnDefinitions>
        <Grid.RowDefinitions>
            <RowDefinition Height="Auto"/>
            <RowDefinition Height="Auto"/>
            <RowDefinition Height="Auto"/>
            <RowDefinition Height="Auto"/>
        </Grid.RowDefinitions>

        <TextBlock Grid.Column="0" Grid.Row="0" Grid.ColumnSpan="2" Margin="4" Text="{Binding FullName}" HorizontalAlignment="Center" FontWeight="Bold"/>

        <Label Grid.Column="0" Grid.Row="1" Margin="4" Content="First Name:" HorizontalAlignment="Right"/>
        
        <TextBox Grid.Column="1" Grid.Row="1" Margin="4" Text="{Binding FirstName, UpdateSourceTrigger=PropertyChanged}" HorizontalAlignment="Left" Width="200"/>

        <Label Grid.Column="0" Grid.Row="2" Margin="4" Content="Last Name:" HorizontalAlignment="Right"/>
        <TextBox Grid.Column="1" Grid.Row="2" Margin="4" Text="{Binding LastName, UpdateSourceTrigger=PropertyChanged}" HorizontalAlignment="Left"  Width="200"/>

        <Label Grid.Column="0" Grid.Row="3" Margin="4" Content="Age:" HorizontalAlignment="Right"/>
        <TextBlock Grid.Column="1" Grid.Row="3" Margin="4" Text="{Binding Age}" HorizontalAlignment="Left"/>
    </Grid>
</Window>
 
 and MaimWindow.xaml.cs
 namespace CommunityToolkit;

/// <summary>
/// Interaction logic for MainWindow.xaml
/// </summary>
public partial class MainWindow : Window
{
    private readonly MyViewModel _viewModel;

    public MainWindow()
    {
        InitializeComponent();
        _viewModel = new MyViewModel();
        // The DataContext serves as the starting point of Binding Paths
        DataContext = _viewModel;
    }
}

What is wrong here ????

**Comment by CADTraveller** [comment, marked as answer]
You need to register the recipient, typically in the constructor; I don't see where you've done this.  Have a look at the docs:

https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/messenger
